dask/typing.py

```diff
@@ -0,0 +1,364 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Callable, Hashable, Mapping, Sequence
+from typing import Any, Protocol, TypeVar, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = Any
+
+
+CollType = TypeVar("CollType", bound="DaskCollection")
+CollType_co = TypeVar("CollType_co", bound="DaskCollection", covariant=True)
+PostComputeCallable = Callable
+
+
+class SchedulerGetCallable(Protocol):
+    """Protocol defining the signature of a __dask_scheduler__ callable."""
+
+    def __call__(
+        self,
+        dask: Mapping,
+        keys: Sequence[Hashable] | Hashable,
+        **kwargs: Any,
+    ) -> Any:
+        """Method called as the default scheduler for a collection.
+
+        Parameters
+        ----------
+        dsk : Mapping
+            The task graph.
+        keys : Sequence[Hashable] | Hashable
+            Key(s) corresponding to the desired data.
+        **kwargs : Any
+            Additional arguments.
+
+        Returns
+        -------
+        Any
+            Result(s) associated with `keys`
+
+        """
+        raise NotImplementedError("Inheriting class must implement this method.")
+
+
+class PostPersistCallable(Protocol[CollType_co]):
+    """Protocol defining the signature of a __dask_postpersist__ callable."""
+
+    def __call__(
+        self,
+        dsk: Mapping,
+        *args: Any,
+        rename: Mapping[str, str] | None = None,
+    ) -> CollType_co:
+        """Method called to rebuild a persisted collection.
+
+        Parameters
+        ----------
+        dsk: Mapping
+            A mapping which contains at least the output keys returned
+            by __dask_keys__().
+        *args : Any
+            Additional optional arguments If no extra arguments are
+            necessary, it must be an empty tuple.
```

Comment:
I think this is not correct. The result of `__dask_postpersist__()` must return an empty tuple, but that corresponds to no `args` here in this callable.